docs/reference/migration/migrate_8_0/security.asciidoc

```diff
@@ -86,6 +107,12 @@ It is now an error to configure any SSL settings for
 `xpack.security.transport.ssl` without also configuring
 `xpack.security.transport.ssl.enabled`.
 
+*Impact* +
+If using `xpack.security.transport.ssl` settings, ensure
+`xpack.security.transport.ssl.enabled` is set as `true`. Not specifying
+`xpack.security.transport.ssl.enabled` as `true` in `elasticsearch.yml` will
+result in an error on startup.
```

Comment:
This isn't quite right. You are fee to set `xpack.security.transport.ssl.enabled` to `true` or `false`, you just can't leave it unspecified if you have set other any `xpack.security.transport.ssl.*` settings. 

This setting defaults to `false` if unspecified. The 7.x (and earlier) behaviour is that nodes that had a `xpack.security.transport.ssl` setting (e.g. `xpack.security.transport.ssl.certificate`) but did not explicitly set `ssl.enabled`, would not have SSL enabled.  
That was confusing because on first inspection it would look like SSL was configured, but since it was not explicitly enabled, it didn't actually do anything.

For people in that situation,  the _no impact_ upgrade would be to explicitly set `ssl.enabled` to `false` to acknowledge that even though they've configured SSL they don't want it to be on. 

The "right fix" is for them to work out why they have `ssl.*` settings in their yml but don't have SSL enabled. Is that intentional?
- If so, remove the `ssl.*` settings _or_ set `ssl.enabled` to false.
- Otherwise, maybe they want to turn on ssl (`ssl.enabled: true`), but that's a bigger change & we can't really walk them through it here (but we could point to the relevant docs).

I don't really know how to explain that well in the breaking changes doc. How much detail do we want to go to?
But I don't think we can recommend people just set it to `true` because that would cause a noticeable change to their cluster - it would turn on SSL when it was previously disabled - and they need to at least think through how they want to coordinate that change across their nodes (for `transport.ssl`) or with clients (for `http.ssl`)